src/tess.py

```diff
@@ -0,0 +1,364 @@
+        
+        # name should be source_id for GAIA catalog entry
+        name = cat_row['name']
+        dec = cat_row['dec']
+        ra = cat_row['ra']
+        mag = cat_row['mag']
+        if mag > 16:
+            # TESS can't see stars fainter than 15 mag
+            print(f"Magnitude of {mag} is too faint for TESS.")
+            return pd.DataFrame()
+
+        cone_radius = kwargs['cone_radius'] \
+                        if 'cone_radius' in kwargs else 0.01
+        sign = ' -' if dec < 0 else ' '
+        coord_str = str(ra) + sign + str(dec)
+        
+        kwargs = {
+            'coordinates': coord_str,
+            'catalog': 'TIC',
+            'radius': cone_radius
+        }
+        catalog_data = self._try_query(Catalogs.query_region, kwargs)
+        if len(catalog_data) == 0:
+            print("No TESS object found for given catalog row.")
+            return pd.DataFrame()
+        
+        # TODO: check for better way of selecting candidate
+        # catalog_data is automatically sorted by dstArcSec, index 0 is closest
+        # also, is GAIAmag guaranteed to exist? 
+        # what if we're not using a GAIA cat_row?
+        candidate = 0
+        while abs(catalog_data['GAIAmag'][candidate] - mag) > MAGDIFF_THRESHOLD:
+            candidate += 1
+        
+        # if there are multiple likely candidates, throw warning
+        close_mags = 0
+        for i in range(len(catalog_data)):
+            gm = catalog_data['GAIAmag'][i]
+            dst = catalog_data['dstArcSec'][i]
+            if abs(gm - mag) < MAGDIFF_THRESHOLD and dst < DST_THRESHOLD:
+                close_mags += 1
+        if close_mags > 1:
+            warnings.warn(f"Multiple ({close_mags}) sources with similar " +
+                            "magnitudes in cone search." +
+                            " Might select incorrect source.")
+
+        TESS_ID = catalog_data['ID'][candidate]
+        kwargs = {
+            'objectname': 'TIC ' + TESS_ID,
+            'radius': 0.1,
+            'obs_collection': 'TESS',
+            # 'dataproduct_type': 'timeseries'
+        }
+        data_query = self._try_query(Observations.query_criteria, kwargs)
+        if len(catalog_data) == 0:
+            print("No data found for object " + 'TIC ' + TESS_ID + '.')
+            return pd.DataFrame()
+        if 'timeseries' not in data_query['dataproduct_type']:
+            print("No timeseries data found for object " + 
+                    'TIC ' + TESS_ID + '.')
+            return pd.DataFrame()
+        
+        lc_indices = []
+        for i, uri in enumerate(data_query['dataURL']):
+            if isinstance(uri, str) and uri[-5:-7] == 'lc':
+                lc_indices.append(i)
+        print(f"Found {len(lc_indices)} light curve(s) for this source.")
+
+        base_url = "https://mast.stsci.edu/api/v0.1/Download/file?uri="
+
+        # how to properly aggregate data if more than 1 LC?
+        # do we just add rows naively?
```

Comment:
Yes, right now my best idea for storing the raw data is just as one big dataframe for each object/observatory. 
We may end up building something better but I think this is fine. 
It is up to the reducer function to split this up by whatever criteria, e.g., by gaps in the data, buy sector ID, etc.